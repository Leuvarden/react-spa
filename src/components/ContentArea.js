import React from 'react';

export default function ContentArea () {
    return (
        <section className="content-container">
            <div className="content__empty">No films found</div>
        </section>
    );
}