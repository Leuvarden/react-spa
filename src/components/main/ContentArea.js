import React, { Component } from 'react';
import ContentItem from './ContentItem';
import EmptyContent from './EmptyContent';
import FilmPanel from './FilmPanel';
import ErrorBoundary from './../Error.js';
import FetchItems from './FetchItems';
import './../../styles/content-area.scss';

export default class ContentArea extends Component {
    constructor(props) {
        super(props);
        this.url = 'http://react-cdp-api.herokuapp.com/';

        this.state = {
            contentData: []
        };

        this.updateFilmPanel = this.updateFilmPanel.bind(this);
    }

    getContent () {
        if (this.state.contentData) {
            const items = (this.state.contentData).map((el) => 
                <ErrorBoundary key={el.id} showOnError={this.getErrorDiv()}>                 
                    <ContentItem
                        key={el.id} 
                        genres={el.genres} 
                        img={el.poster_path}
                        date={el.release_date} 
                        title={el.title} 
                        overview={el.overview}
                        updateFilmPanel={this.updateFilmPanel}
                    />
                </ErrorBoundary>
            );
            return items;
        } else {
            return <EmptyContent />;
        }
    }

    getPanel() {
        if (!this.state.currentItem) {
            return;
        } else {
            let item = this.state.currentItem;
            return <FilmPanel 
                img={item.img}
                title={item.title}
                genres={item.genres}
                overview={item.overview} 
                date={item.date}
            />;
        }
    }

    render() {   
        // console.log(this.state); 
        return (
            <section className='content-container'>
                {this.getPanel()}
                {this.getContent()}
            </section>
        );
    }

    componentDidMount() {
        FetchItems(`${this.url}movies`)
            .then(response => this.setState({contentData: response.data }) );
    }

    updateFilmPanel(filmProps) {
        this.setState({currentItem: filmProps});
    }

    getErrorDiv () {
        return (
            <figure className="content-item">
                <img src='./../../public/img/cube.gif' className="content-item_img"></img>
                <figcaption className="content-item__description">
                    <span className="content-item_title"><b>Loading...</b></span>
                </figcaption>
            </figure>
        );
    }
}
