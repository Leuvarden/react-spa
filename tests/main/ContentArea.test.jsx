import React from 'react';
import ContentArea from './../../src/components/main/ContentArea.js';

/* eslint-disable no-undef */

const content = [
    {
        id: 1,
        genres: [ 'Adventure', 'Science'],
        img: 'poster_path',
        date: '2014-05-02',
        title:'Pulp Fiction',
        overview: 'overview'
    }, 
    {
        id: 2,
        genres: [ 'Adventure', 'Science'],
        img: 'poster_path',
        date: '2014-05-02',
        title:'Pulp Fiction',
        overview: 'overview'
    }
];

describe ('ContentArea', () => {

    it('renders as expected', () => {
        const main = shallow(
            <ContentArea />);
    
        expect(main).toMatchSnapshot();
    });

    // it('adds content items', () => {

    //     // const spy = jest.spyOn(ContentArea, 'getContent');

    //     const main = shallow(<ContentArea />);

    //     // expect(main.find('.content-item').length).toBe(0);

    //     main.setState({
    //         contentData: content
    //     });

    //     // console.log(main)

    //     // expect(main.find('.content-item').length).toBe(2);

    //     // main.unmount();

    //     // expect(spy).toHaveBeenCalled();
    // });

});