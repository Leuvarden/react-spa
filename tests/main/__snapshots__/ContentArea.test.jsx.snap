// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentArea renders as expected 1`] = `
<section
  className="content-container"
/>
`;
